underfs/b2/src/main/java/alluxio/underfs/b2/B2UnderFileSystem.java

```diff
@@ -0,0 +1,374 @@
+      String accountId) {
+    super(alluxioURI, conf);
+    requireNonNull(alluxioURI, "alluxioURI is null");
+    requireNonNull(conf, "conf is null");
+    requireNonNull(b2StorageClient, "b2StorageClient is null");
+    requireNonNull(bucketId, "bucketId is null");
+    requireNonNull(bucketName, "bucketName is null");
+    requireNonNull(bucketMode, "bucketMode is null");
+    requireNonNull(accountId, "accountId is null");
+
+    LOG.debug("alluxioURI: {}", alluxioURI);
+    LOG.debug("buckedId: {}", bucketId);
+    LOG.debug("bucketName: {}", bucketName);
+    LOG.debug("bucketMode: {}", bucketMode);
+    LOG.debug("accountId: {}", accountId);
+
+    mB2StorageClient = b2StorageClient;
+    mBucketId = bucketId;
+    mBucketName = bucketName;
+    mBucketMode = bucketMode;
+    mAccountId = accountId;
+    mFolderIndicator = conf.get(PropertyKey.B2_FOLDER_INDICATOR);
+  }
+
+  @Override
+  protected boolean copyObject(String src, String dst) throws IOException {
+    LOG.debug("copyObject src: {}, dst: {}", src, dst);
+    // source request
+    B2ListFileVersionsRequest srcRequest =
+        B2ListFileVersionsRequest.builder(mBucketId).setStartFileName(src).setPrefix(src).build();
+    B2FileVersion srcFileVersion = null;
+    try {
+      for (B2FileVersion version : mB2StorageClient.fileVersions(srcRequest)) {
+        if (version.getFileName().equals(src)) {
+          srcFileVersion = version;
+        } else {
+          break;
+        }
+      }
+    } catch (B2Exception e) {
+      e.printStackTrace();
+    }
+    if (srcFileVersion == null) {
+      LOG.error("Source object not found: {}", src);
+      return false;
+    }
+    String sourceFileId = srcFileVersion.getFileId();
+    // destination request
+    B2ListBucketsRequest bucketsRequest =
+        B2ListBucketsRequest.builder(mAccountId).setBucketName(dst).build();
+    try {
+      mB2StorageClient.listBuckets(bucketsRequest);
+    } catch (B2Exception e) {
+      e.printStackTrace();
+    }
+    getExistingDirectoryStatus(dst);
+    B2CopyFileRequest copyFileRequest =
+        B2CopyFileRequest.builder(sourceFileId, src).setDestinationBucketId(mBucketId).build();
+    try {
+      mB2StorageClient.copySmallFile(copyFileRequest);
+    } catch (B2Exception e) {
+      e.printStackTrace();
+    }
+
+    return true;
+  }
+
+  @Override
+  public boolean createEmptyObject(String key) {
+    final File mFile = new File(PathUtils
+        .concatPath(CommonUtils.getTmpDir(mUfsConf.getList(PropertyKey.TMP_DIRS, ",")),
+            UUID.randomUUID()));
+    final B2ContentSource source = B2FileContentSource.builder(mFile).build();
+
+    B2UploadFileRequest request =
+        B2UploadFileRequest.builder(mBucketName, key, B2ContentTypes.APPLICATION_OCTET, source)
+            .build();
+
+    try {
+      mB2StorageClient.uploadSmallFile(request);
+      return true;
+    } catch (B2Exception e) {
+      LOG.error("Failed to create an empty object on B2with key:{}", key);
```

Comment:
can we also add the exception to the log message, so that we can see the exception and stacktrace?
```suggestion
      LOG.error("Failed to create an empty object on B2with key: {}", key, e);
```